### PR TITLE
Move click handler outside PureComponent render

### PR DESCRIPTION
**Pull request by treyhuffine**
If I understand correctly, won't this cause a re-render every time since we're creating a new function inside the render method? Moving it outside to prevent this.

### CODE REVIEW

**Comment by aweary** [comment]
Thanks for the PR @treyhuffine!

> If I understand correctly, won't this cause a re-render every time since we're creating a new function inside the render method? Moving it outside to prevent this.

This can be true if you're passing a function to another custom component that's using a shallow compare in `shouldComponentUpdate` to prevent re-renders. In this example, the function is being passed a DOM element, `button`, not another custom component. That means there's no subtree that will or won't re-render since `button` is a leaf node in the render tree.

Now, if you were passing a new function to this `CounterButton` every time

```jsx
<CounterButton onClick={(event) => {...} />
```

You might have a problem since this component is extending `PureComponent` so it would re-render when it didn't need to.

In other words, when this `render` method is called it's going to attempt to re-render the button no matter what. React is smart enough to know whether it needs to touch the DOM or not. In this case, it might remove the old listener and add the new one, but that's not a big deal thanks to React's synthetic event system. The tradeoff here is that this syntax is more succinct which lets the example be more focused.

I hope that cleared things up a bit!
